### PR TITLE
Try using Editor in Docs

### PR DESCRIPTION
**Pull request by aurindam**
Now you can edit the code in the docs and press on "play" button to see the changes. There is also a copy button for easily copying the code.

### CODE REVIEW

**Comment by tronical** [comment]
I get a JavaScript error when trying this: `Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'cm6.createEditorState')`.

I assume that `cm6` refers to CodeMirror as editor. I'm not in favor of shipping minified bundles in our repo, it's like embedding binaries. This might mean that it may be necessary to add a bundler to our sphinx build.

File: .github/workflows/upgrade_version.yaml [hunk 1]
```diff
@@ -42,6 +42,9 @@ jobs:
                   # Version in the docs (cargo add slint@<VERSION>):
                   git ls-files | grep "\(^\|/\)docs/.*\.\(md\|rst\)\$" | xargs sed -i 's/slint@[0-9]\+\.[0-9]\+\.[0-9]\+/slint@${{ github.event.inputs.new_version }}/'
 
+                  # Version in the docs (codemirror.js and rollup.config.js):
+                  sed -i 's/[0-9]\+\.[0-9]\+\.[0-9]\+/${{ github.event.inputs.new_version }}/' docs/editor/codemirror.js docs/editor/rollup.config.js
```
Comment (ogoffart) [inline .github/workflows/upgrade_version.yaml:46]: I don't see the version number in these files.
Is there perhaps another way to get the version?
Should it perhaps be bart of the release script instead like for slintpad?

Comment (aurindam) [reply .github/workflows/upgrade_version.yaml:46]: Good catch. This is not needed any longer. Its 'fixed' in build_docs.yaml

sed -i "s/snapshots.slint.dev\/master/releases.slint.dev\/$version/" docs/editor/codemirror.js docs/editor/rollup.config.js

File: docs/reference/conf.py [hunk 2]
```diff
@@ -119,6 +121,9 @@
 rst_epilog = """
 """
 
+from slint_translator import SlintHTML5Translator
 
 def setup(app):
-    app.add_css_file("theme_tweak.css")
+    # Set the custom HTML translator, overriding the default one
+    print("Custom translator setup is being called.")
```
Comment (tronical) [inline docs/reference/conf.py:128]:
```suggestion
```

**Review by tronical — APPROVED** [review]

**Pull request merged**